libraries/AP_Baro/AP_Baro_BMP085.cpp

```diff
@@ -138,17 +223,24 @@ void AP_Baro_BMP085::_cmd_read_pressure()
 bool AP_Baro_BMP085::_read_pressure()
 {
     uint8_t buf[3];
+    if (_dev->read_registers(0xF6, buf, sizeof(buf))) {
+        _raw_pressure = (((uint32_t)buf[0] << 16)
+                         | ((uint32_t)buf[1] << 8)
+                         | ((uint32_t)buf[2])) >> (8 - OVERSAMPLING);
+        return true;
+    }
 
-    if (!_dev->read_registers(0xF6, buf, sizeof(buf))) {
-        _retry_time = AP_HAL::millis() + 1000;
-        _dev->set_speed(AP_HAL::Device::SPEED_LOW);
-        return false;
+    uint8_t xlsb;
+    if (_dev->read_registers(0xF6, buf, 2) && _dev->read_registers(0xF8, &xlsb, 1) ) {
+        _raw_pressure = (((uint32_t)buf[0] << 16)
+                       | ((uint32_t)buf[1] << 8)
+                       | ((uint32_t)xlsb)) >> (8 - OVERSAMPLING);
+        return true;
     }
 
-    _raw_pressure = (((uint32_t)buf[0] << 16)
-                     | ((uint32_t)buf[1] << 8)
-                     | ((uint32_t)buf[2])) >> (8 - OVERSAMPLING);
-    return true;
+    _last_press_read_command_time = AP_HAL::millis() + 1000;
```

Comment:
this variable name does not match how you use it. "last" would mean now without the offset. You're using it as if it says "next".

Reply:
yes you are right

Reply:
sorry I missed. This is really LAST time, but now time by force shifted to the future to give a gap